### ISSUE TITLE
Questions on Linux usage

### ISSUE THREAD

**Issue by sunny137** [frgtn/rpcapd-linux#5]
Hi, thanks for the port. I have been looking for a tool like this for a long time on Linux. Few queries:

1. I could compile it on Ubuntu 17.04 by linking it directly to systems libpcap. I see that you have a patched version of libpcap, is this required? Would this work if I directly link to system libpcap?

2. I saw the disclaimers but do you have know of any production usage to passively capture traffic and relay it to other hosts?

Any inputs will be appreciated. This is exactly the tool I am looking for but not sure about the stability and support etc.

**Comment by frgtn** [comment]
Hi @sunny137. First of all I must say that this is not so much of a port, it's based on some blog posts I found about compiling rpcapd on Linux, placed conveniently in a repo instead of a set of instructions. I haven't linked to the blog post(s) at the time and can't find them now.

Answers to your questions:
1. I don't think there should be any problems with linking to system libpcap. I haven't tested it myself though.
2. This "port" is not actively maintained, unfortunately. I used it for a single project back in 2012 when we were testing embedded WiFi implementation and needed a good network sensor that multiple team members could use when testing / debugging.

For production use I suggest looking at other tools which are designed specifically for capturing / inspecting network traffic at scale and have better operability. Unfortunately I don't have any good suggestions.

I'm closing this issue as I think the questions have been answered, feel free to reopen it if you have further inquiries.

**Comment by sunny137** [comment]
Yes, I am opening this since I have couple more queries. I could get this compiled against system libpcap on Ubuntu 16.04, so this part is good.

I am just not able to wrap my head around the actual usage of this tool when the I have something apart from wireshark at the other end. E.g. Is it possible to sniff packets passively using rpcapd on machine 1 and send those packets out to say snort running on a different machine? Does it require snort to understand rpcap protocol, decode it and then extract actual pcap ?

Example:

I want to capture traffic on port 80 on VM1, I have nginx serving on port 80. After capturing this traffic, I want to send this out to VM2 running snort for further analysis. Is this possible using rpcapd? Would help me a great deal to see this actually working. All I end up with is 

Is the server properly installed on x.x.x.x?  connect() failed: Connection refused (code 111) 

Just instructions to grab packets on one VM1 and somehow reading them at the other end (using say tcpdump) would also help me debug this further.

**Comment by sunny137** [comment]
I am not able to re-open this issue.

**Comment by frgtn** [comment]
I initially used this tool with Wireshark as a remote sensor, where the daemon would run on a Linux machine with WiFi cards that could monitor and Windows users would connect to using Wireshark when doing their testing.

rpcapd is originally from WinPCAP project and they have some docs about it: https://www.winpcap.org/docs/docs_40_2/html/group__remote.html It does seem that it could potentially be used with tools other than Wireshark. There is another fork of rpcapd that may contain more useful information: https://github.com/ExtraHop/rpcapd

Regarding your example, did you try to connect to your VM with Wireshark? Given that you're seeing connection refused, did you check with `sudo netstat -pnl` on VM1 that rpcapd is actually listening on a port?

One thing to watch out for is using same interface for capture and communication between rpcapd and Wireshark as rpcapd will capture its own traffic potentially DoSing itself. It may be possible to set up rules to ignore rpcapd traffic but I don't know how to do that out of the top of my head.

Hope this helps a little. Also, if you end up using it and would like to take over maintenance I'd be more than happy to pass it to you as I haven't been using this tool for quite a while now.

**Comment by guyharris** [comment]
Try using the standard libpcap client and server code in [its GitHub repository](https://github.com/the-tcpdump-group/libpcap); file any issues found with that code in its Issues section.
